torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -272,6 +272,19 @@ InputsIdLookup::IdLookupReturn InputsIdLookup::lookupId(
 FusionExecutorCache::FusionExecutorCache(std::unique_ptr<Fusion>&& fusion)
     : fusion_(std::move(fusion)) {
   FUSER_PERF_SCOPE("FusionExecutorCache::FusionExecutorCache");
+
+  // case of segmented fusion
+  // TODO: might be worthwhile re-using the SchedulerEntry infrastructure for
+  //       single-kernel fusion as well.
+  const bool can_schedule =
+      SchedulerEntry::proposeHeuristics(fusion_.get()).has_value();
+  const bool segmented = !can_schedule;
```

Comment:
why do we need two variables here?

Reply:
Right we only need one here. Updated, thanks!